pkg/sql/distsqlrun/server.go

```diff
@@ -148,6 +151,13 @@ type ServerConfig struct {
 	// A handle to gossip used to broadcast the node's DistSQL version and
 	// draining state.
 	Gossip *gossip.Gossip
+
+	// SessionBoundInternalExecutorFactory is used to construct session-bound
+	// executors. The idea is that a higher-layer binds some of the arguments
+	// required, so that users of ServerConfig don't have to care about them.
```

Comment:
r6: The idiomatic go name for this callback would be `MakeSessionBoundInternalExecutor`. Then calls to it will be much more pleasant to read.